### PR TITLE
Support namespaced exports

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: contrib/example-backend/controllers/clusterbinding/clusterbinding_controller.go [hunk 1]
```diff
@@ -114,6 +176,45 @@ func (c *Controller) enqueueServiceExport(logger klog.Logger, obj interface{}) {
 	c.queue.Add(key)
 }
 
+func (c *Controller) enqueueServiceExport(logger klog.Logger, obj interface{}) {
+	seKey, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
```
Comment (s-urbaniak) [inline contrib/example-backend/controllers/clusterbinding/clusterbinding_controller.go:210]: curiosity question: why not simply `MetaNamespaceKeyFunc(obj)` ?

File: contrib/example-backend/controllers/clusterbinding/clusterbinding_reconcile.go [hunk 2]
```diff
@@ -18,51 +18,187 @@ package clusterbinding
...
+	if clusterBinding.Status.LastHeartbeatTime.IsZero() {
+		conditions.MarkFalse(clusterBinding,
 			kubebindv1alpha1.ClusterBindingConditionHealthy,
 			"FirstHeartbeatPending",
 			conditionsapi.ConditionSeverityInfo,
 			"Waiting for first heartbeat",
 		)
-	} else if export.Status.HeartbeatInterval.Duration == 0 {
-		conditions.MarkFalse(export,
+	} else if clusterBinding.Status.HeartbeatInterval.Duration == 0 {
```
Comment (s-urbaniak) [inline contrib/example-backend/controllers/clusterbinding/clusterbinding_reconcile.go:76]: much better naming than export indeed 👍

File: contrib/example-backend/options/options.go [hunk 3]
```diff
@@ -78,6 +82,7 @@ func (options *Options) AddFlags(fs *pflag.FlagSet) {
 	fs.StringVar(&options.KubeConfig, "kubeconfig", options.KubeConfig, "path to a kubeconfig. Only required if out-of-cluster")
 	fs.StringVar(&options.NamespacePrefix, "namespace-prefix", options.NamespacePrefix, "The prefix to use for cluster namespaces")
 	fs.StringVar(&options.PrettyName, "pretty-name", options.PrettyName, "Pretty name for the backend")
+	fs.StringVar(&options.ConsumerScope, "consumer-scope", options.ConsumerScope, "How consumers access the service provider cluster. In Kubernetes Namespaced allows namespace isolation. In kcp Cluster allows workspace isolation, and with that allows cluster-scoped resources to bind and it is generally more performant.")
```
Comment (s-urbaniak) [inline contrib/example-backend/options/options.go:85]: Suggestions:
- allow lowercase for `namespaced` and `cluster`.

CLI documentation suggestion:
> How consumers access the service provider cluster. Supported scopes are "namespaced" for namespace based isolation if service provider clusters are using Kubernetes and "cluster" for workspace based isolation if service provider clusters are using kcp.

File: contrib/example-backend/controllers/clusterbinding/clusterbinding_controller.go (hunk 1 above)
Comment (sttts) [reply contrib/example-backend/controllers/clusterbinding/clusterbinding_controller.go:222]: I want to get the key if the object was lost due to informer restart, i.e. obj is a tombstone

**Review by s-urbaniak — APPROVED** [review]

**Pull request merged**